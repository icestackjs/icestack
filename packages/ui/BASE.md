<br>

<p align="center">
<img src="https://github.com/sonofmagic/icestack/raw/main/assets/logo.svg" style="width:100px;" />
</p>

<h1 align="center">IceStack</h1>

❤️ IceStack, Web UI for Mobile and PC , open-source Headless component library generator

## Documentation

[Documentation](https://ui.icebreaker.top/) | [中文文档](https://ui.icebreaker.top/zh-CN)

{{replace}}
