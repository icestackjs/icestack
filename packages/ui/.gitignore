assets/css
assets/plugins
assets/js
assets/css-resolved