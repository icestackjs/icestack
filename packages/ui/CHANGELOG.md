# @icestack/ui

## 1.3.3

### Patch Changes

- [`e72e248`](https://github.com/sonofmagic/icestack/commit/e72e2483416f9d10401f8d2dfb991c158c3d2cce) Thanks [@sonofmagic](https://github.com/sonofmagic)! - add comment to cva

- Updated dependencies [[`e72e248`](https://github.com/sonofmagic/icestack/commit/e72e2483416f9d10401f8d2dfb991c158c3d2cce)]:
  - @icestack/cli@1.3.3
  - @icestack/config@1.3.3
  - @icestack/core@1.3.3
  - @icestack/logger@1.3.3
  - @icestack/postcss-utils@1.3.3
  - @icestack/preset-default@1.3.3
  - @icestack/scss@1.3.3
  - @icestack/shared@1.3.3
  - @icestack/theme-algorithm@1.3.3
  - @icestack/types@1.3.3

## 1.3.2

### Patch Changes

- [`8aafb36`](https://github.com/sonofmagic/icestack/commit/8aafb36725359619bf94a40079b387d747ff02a5) Thanks [@sonofmagic](https://github.com/sonofmagic)! - fix @icestack/preset-default

- Updated dependencies [[`8aafb36`](https://github.com/sonofmagic/icestack/commit/8aafb36725359619bf94a40079b387d747ff02a5)]:
  - @icestack/cli@1.3.2
  - @icestack/config@1.3.2
  - @icestack/core@1.3.2
  - @icestack/logger@1.3.2
  - @icestack/postcss-utils@1.3.2
  - @icestack/preset-default@1.3.2
  - @icestack/scss@1.3.2
  - @icestack/shared@1.3.2
  - @icestack/theme-algorithm@1.3.2
  - @icestack/types@1.3.2

## 1.3.1

### Patch Changes

- [`dc76f5c`](https://github.com/sonofmagic/icestack/commit/dc76f5c4c3f06aba8f2d0f8982d48ddadf7ec741) Thanks [@sonofmagic](https://github.com/sonofmagic)! - support sass @use and @import

- Updated dependencies [[`dc76f5c`](https://github.com/sonofmagic/icestack/commit/dc76f5c4c3f06aba8f2d0f8982d48ddadf7ec741)]:
  - @icestack/cli@1.3.1
  - @icestack/config@1.3.1
  - @icestack/core@1.3.1
  - @icestack/logger@1.3.1
  - @icestack/postcss-utils@1.3.1
  - @icestack/preset-default@1.3.1
  - @icestack/scss@1.3.1
  - @icestack/shared@1.3.1
  - @icestack/theme-algorithm@1.3.1
  - @icestack/types@1.3.1

## 1.3.0

### Minor Changes

- [`4647129`](https://github.com/sonofmagic/icestack/commit/4647129503b4f4872a354772d65daf8a42403594) Thanks [@sonofmagic](https://github.com/sonofmagic)! - release postcss usage

### Patch Changes

- Updated dependencies [[`4647129`](https://github.com/sonofmagic/icestack/commit/4647129503b4f4872a354772d65daf8a42403594)]:
  - @icestack/cli@1.3.0
  - @icestack/config@1.3.0
  - @icestack/core@1.3.0
  - @icestack/logger@1.3.0
  - @icestack/postcss-utils@1.3.0
  - @icestack/preset-default@1.3.0
  - @icestack/scss@1.3.0
  - @icestack/shared@1.3.0
  - @icestack/theme-algorithm@1.3.0
  - @icestack/types@1.3.0

## 1.2.0

### Minor Changes

- [`7706a6a`](https://github.com/sonofmagic/icestack/commit/7706a6aa6ef786308612e9680104fcc485cdd6d0) Thanks [@sonofmagic](https://github.com/sonofmagic)! - add build cache and postcss plugin

### Patch Changes

- Updated dependencies [[`7706a6a`](https://github.com/sonofmagic/icestack/commit/7706a6aa6ef786308612e9680104fcc485cdd6d0)]:
  - @icestack/cli@1.2.0
  - @icestack/config@1.2.0
  - @icestack/core@1.2.0
  - @icestack/logger@1.2.0
  - @icestack/postcss-utils@1.2.0
  - @icestack/preset-default@1.2.0
  - @icestack/scss@1.2.0
  - @icestack/shared@1.2.0
  - @icestack/theme-algorithm@1.2.0
  - @icestack/types@1.2.0

## 1.0.3

### Patch Changes

- [`7475342`](https://github.com/sonofmagic/icestack/commit/747534263ba1dba873afc8bd409160e189a75e54) Thanks [@sonofmagic](https://github.com/sonofmagic)! - deps on @icestack config and add more colors types

- Updated dependencies [[`7475342`](https://github.com/sonofmagic/icestack/commit/747534263ba1dba873afc8bd409160e189a75e54)]:
  - @icestack/config@1.0.0
  - @icestack/logger@0.6.3
  - @icestack/postcss@1.0.1
  - @icestack/preset-default@1.0.1
  - @icestack/scss@1.0.1
  - @icestack/shared@1.0.1
  - @icestack/theme-algorithm@1.0.1
  - @icestack/types@1.0.1

## 1.0.2

### Patch Changes

- [`1f01400`](https://github.com/sonofmagic/icestack/commit/1f0140087d23d993511a36f189920ed6db519f04) Thanks [@sonofmagic](https://github.com/sonofmagic)! - fix mediaDarkTheme issue

## 1.0.1

### Patch Changes

- [`53ad0bf`](https://github.com/sonofmagic/icestack/commit/53ad0bfb8aec4c3901e0f403715d54e2d3784c6d) Thanks [@sonofmagic](https://github.com/sonofmagic)! - fix: custom.js ref error

## 1.0.0

### Major Changes

- [#25](https://github.com/sonofmagic/icestack/pull/25) [`b7f2ee4`](https://github.com/sonofmagic/icestack/commit/b7f2ee41c516b63f6093f5c92fc2375817438347) Thanks [@sonofmagic](https://github.com/sonofmagic)! - chore: 1.0.0

### Patch Changes

- Updated dependencies [[`b7f2ee4`](https://github.com/sonofmagic/icestack/commit/b7f2ee41c516b63f6093f5c92fc2375817438347)]:
  - @icestack/preset-default@1.0.0
  - @icestack/postcss@1.0.0
  - @icestack/shared@1.0.0
  - @icestack/types@1.0.0
  - @icestack/scss@1.0.0
  - @icestack/logger@0.6.2

## 0.6.2

### Patch Changes

- [`854af96`](https://github.com/sonofmagic/icestack/commit/854af960284af72cbf36ebaf6e574c267dea42b1) Thanks [@sonofmagic](https://github.com/sonofmagic)! - add unocss types

- Updated dependencies [[`854af96`](https://github.com/sonofmagic/icestack/commit/854af960284af72cbf36ebaf6e574c267dea42b1)]:
  - @icestack/logger@0.6.2
  - @icestack/postcss@0.6.2
  - @icestack/preset-default@0.1.2
  - @icestack/shared@0.6.2
  - @icestack/types@0.6.2

## 0.6.1

### Patch Changes

- [`e89ae02`](https://github.com/sonofmagic/icestack/commit/e89ae02b3d0a826a365e97956923b02429fc74ad) Thanks [@sonofmagic](https://github.com/sonofmagic)! - add cli and fix mode preset issue

- Updated dependencies [[`e89ae02`](https://github.com/sonofmagic/icestack/commit/e89ae02b3d0a826a365e97956923b02429fc74ad)]:
  - @icestack/logger@0.6.1
  - @icestack/postcss@0.6.1
  - @icestack/preset-default@0.1.1
  - @icestack/shared@0.6.1
  - @icestack/types@0.6.1

## 0.6.0

### Minor Changes

- [`3311043`](https://github.com/sonofmagic/icestack/commit/33110430aa11e6bd2b58ffcdbd7434df4814c272) Thanks [@sonofmagic](https://github.com/sonofmagic)! - add @icestack/preset-default, change default preset to this package

### Patch Changes

- Updated dependencies [[`3311043`](https://github.com/sonofmagic/icestack/commit/33110430aa11e6bd2b58ffcdbd7434df4814c272)]:
  - @icestack/logger@0.6.0
  - @icestack/postcss@0.6.0
  - @icestack/preset-default@0.1.0
  - @icestack/shared@0.6.0
  - @icestack/types@0.6.0

## 0.5.5

### Patch Changes

- [`05c583d`](https://github.com/sonofmagic/icestack/commit/05c583d0683a87069fa9777c847c2784de4336cf) Thanks [@sonofmagic](https://github.com/sonofmagic)! - support raw string in css schema

- Updated dependencies [[`05c583d`](https://github.com/sonofmagic/icestack/commit/05c583d0683a87069fa9777c847c2784de4336cf)]:
  - @icestack/logger@0.5.5
  - @icestack/postcss@0.5.5
  - @icestack/shared@0.5.5
  - @icestack/types@0.5.5

## 0.5.4

### Patch Changes

- [`22b607d`](https://github.com/sonofmagic/icestack/commit/22b607d939d2a7c1794f179c6fffed41c7436efb) Thanks [@sonofmagic](https://github.com/sonofmagic)! - add pick options and change mode option

- Updated dependencies [[`22b607d`](https://github.com/sonofmagic/icestack/commit/22b607d939d2a7c1794f179c6fffed41c7436efb)]:
  - @icestack/logger@0.5.4
  - @icestack/postcss@0.5.4
  - @icestack/shared@0.5.4
  - @icestack/types@0.5.4

## 0.5.3

### Patch Changes

- [`f48c139`](https://github.com/sonofmagic/icestack/commit/f48c1390111741f21d98c2d8f5117e8ef0d5d080) Thanks [@sonofmagic](https://github.com/sonofmagic)! - add clean options

- Updated dependencies [[`f48c139`](https://github.com/sonofmagic/icestack/commit/f48c1390111741f21d98c2d8f5117e8ef0d5d080)]:
  - @icestack/logger@0.5.3
  - @icestack/postcss@0.5.3
  - @icestack/shared@0.5.3
  - @icestack/types@0.5.3

## 0.5.2

### Patch Changes

- [`99ebaf4`](https://github.com/sonofmagic/icestack/commit/99ebaf459d0727f7f57111827362456f2a9f3f27) Thanks [@sonofmagic](https://github.com/sonofmagic)! - fix: base.mediaDarkTheme option

- Updated dependencies [[`99ebaf4`](https://github.com/sonofmagic/icestack/commit/99ebaf459d0727f7f57111827362456f2a9f3f27)]:
  - @icestack/logger@0.5.2
  - @icestack/postcss@0.5.2
  - @icestack/shared@0.5.2
  - @icestack/types@0.5.2

## 0.5.1

### Patch Changes

- [`cd4a6f0`](https://github.com/sonofmagic/icestack/commit/cd4a6f020fea5164011476f61385322a593f994f) Thanks [@sonofmagic](https://github.com/sonofmagic)! - compose postcss options into one

- Updated dependencies [[`cd4a6f0`](https://github.com/sonofmagic/icestack/commit/cd4a6f020fea5164011476f61385322a593f994f)]:
  - @icestack/logger@0.5.1
  - @icestack/postcss@0.5.1
  - @icestack/shared@0.5.1
  - @icestack/types@0.5.1

## 0.5.0

### Minor Changes

- [`bcb03ca`](https://github.com/sonofmagic/icestack/commit/bcb03ca5b64672e0467ea047cb8c546a0ad57091) Thanks [@sonofmagic](https://github.com/sonofmagic)! - release 0.5.0

### Patch Changes

- Updated dependencies [[`bcb03ca`](https://github.com/sonofmagic/icestack/commit/bcb03ca5b64672e0467ea047cb8c546a0ad57091)]:
  - @icestack/logger@0.5.0
  - @icestack/postcss@0.5.0
  - @icestack/shared@0.5.0
  - @icestack/types@0.5.0

## 0.4.11

### Patch Changes

- [`7a5e4ff`](https://github.com/sonofmagic/icestack/commit/7a5e4ff8f0be55f5377c6c28f11c8badb6f6246a) Thanks [@sonofmagic](https://github.com/sonofmagic)! - ci changes

- Updated dependencies [[`7a5e4ff`](https://github.com/sonofmagic/icestack/commit/7a5e4ff8f0be55f5377c6c28f11c8badb6f6246a)]:
  - @icestack/logger@0.4.11
  - @icestack/postcss@0.4.11
  - @icestack/shared@0.4.11
  - @icestack/types@0.4.11

## 0.4.10

### Patch Changes

- changelog init

- Updated dependencies []:
  - @icestack/logger@0.4.10
  - @icestack/postcss@0.4.10
  - @icestack/shared@0.4.10
  - @icestack/types@0.4.10
