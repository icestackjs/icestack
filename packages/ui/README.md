# @icestack/ui

❤️ IceStack, Web UI for Mobile and PC , open-source Headless component library generator

![logo](https://github.com/sonofmagic/icestack/blob/main/assets/logo.svg)

## How to use?

[Documentation](https://ui.icebreaker.top/)

[中文文档)](https://ui.icebreaker.top/zh-CN)

[Storybook/组件演示](https://story.ui.icebreaker.top/)
