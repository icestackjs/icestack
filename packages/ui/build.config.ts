import path from 'node:path'
import { defineBuildConfig } from 'unbuild'

export default defineBuildConfig({
  entries: ['./src/index', './src/tailwindcss', './src/cli.ts'], //  './src/unocss.ts', //, './src/colors.ts'],
  rollup: {
    // 内联，相当于 nodeResolve
    inlineDependencies: true,
    // cjs
    emitCJS: true,
    // 添加 cjs 注入
    cjsBridge: true,
    dts: {
      // https://github.com/unjs/unbuild/issues/135
      respectExternal: false
    },
    commonjs: {
      ignore: (id) => {
        return id.startsWith('../assets/')
      }
    }
    // resolve: {
    //   resolveOnly: (module) => !module.startsWith('../assets/')
    // }
    // output: {
    //   manualChunks(id) {
    //     if (id.includes('colors')) {
    //       return 'colors'
    //     }
    //   }
    // }
  },
  alias: {
    // 别名
    '@': path.resolve(__dirname, './src')
  },
  // dts
  declaration: true,
  externals: [/^tailwindcss/, /^\.\.\/assets\//, /^unocss/]
})
