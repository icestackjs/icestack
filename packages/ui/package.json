{
  "name": "@icestack/ui",
  "version": "1.3.7",
  "description": "❤️ IceStack, Web UI for Mobile, PC and Mini Program, open-source Headless component library generator",
  "sideEffects": false,
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/index.mjs",
      "require": "./dist/index.cjs"
    },
    "./defaults": {
      "types": "./dist/defaults.d.ts",
      "import": "./dist/defaults.mjs",
      "require": "./dist/defaults.cjs"
    },
    "./components": {
      "types": "./dist/components.d.ts",
      "import": "./dist/components.mjs",
      "require": "./dist/components.cjs"
    },
    "./postcss": {
      "types": "./dist/postcss.d.ts",
      "import": "./dist/postcss.mjs",
      "require": "./dist/postcss.cjs"
    },
    "./config": {
      "types": "./dist/config.d.ts",
      "import": "./dist/config.mjs",
      "require": "./dist/config.cjs"
    },
    "./shared": {
      "types": "./dist/shared.d.ts",
      "import": "./dist/shared.mjs",
      "require": "./dist/shared.cjs"
    },
    "./types": {
      "types": "./dist/types.d.ts",
      "import": "./dist/types.mjs",
      "require": "./dist/types.cjs"
    },
    "./*": "./*"
  },
  "typesVersions": {
    "*": {
      "*": [
        "./dist/*",
        "./dist/index.d.ts"
      ]
    }
  },
  "main": "./dist/index.cjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "files": [
    "bin",
    "dist"
  ],
  "bin": {
    "icestack": "bin/icestack.js"
  },
  "scripts": {
    "build": "npm run build:plugin",
    "build:all": "node scripts/build-scss.js",
    "dev": "npm run clean && npm run build:all && unbuild --sourcemap",
    "bp": "unbuild",
    "build:plugin": "unbuild",
    "clean": "node scripts/clean.js",
    "lint": "eslint --cache --ext .ts,.js,.mjs,.cjs . && prettier -c src test",
    "lint:fix": "eslint --cache --ext .ts,.js,.mjs,.cjs . --fix && prettier -c src test -w",
    "prepack": "pnpm run build",
    "release": "pnpm test && changelogen --release && npm publish && git push --follow-tags",
    "test": "vitest run",
    "test:dev": "vitest",
    "get-table": "node scripts/get-table.js",
    "get-group": "node scripts/get-group.js",
    "get-svg": "node scripts/svg64.js",
    "static": "npm run get-table && npm run get-group",
    "tailwind-config-viewer": "tailwind-config-viewer -o",
    "raw": "node bin/icestack.js inspect -o ./all0.json",
    "sync": "cnpm sync @icestack/ui",
    "watch": "node scripts/watch.js"
  },
  "dependencies": {
    "@icestack/core": "workspace:^",
    "@icestack/config": "workspace:^",
    "@icestack/logger": "workspace:^",
    "@icestack/postcss-utils": "workspace:^",
    "@icestack/preset-default": "workspace:^",
    "@icestack/scss": "workspace:^",
    "@icestack/shared": "workspace:^",
    "@icestack/theme-algorithm": "workspace:^",
    "@icestack/types": "workspace:^",
    "@icestack/cli": "workspace:^"
  },
  "devDependencies": {
    "@types/lodash": "^4.14.202"
  },
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "author": "SonOfMagic <qq1324318532@gmail.com>",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/sonofmagic/icestack"
  },
  "homepage": "https://ui.icebreaker.top",
  "repository": {
    "type": "git",
    "url": "https://github.com/sonofmagic/icestack"
  },
  "keywords": [
    "ui",
    "extend",
    "css",
    "generator",
    "tailwindcss",
    "unocss",
    "components",
    "ui-library",
    "postcss",
    "build"
  ]
}
