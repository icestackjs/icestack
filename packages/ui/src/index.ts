import { CodegenOptions, DeepPartial } from './types'
export * from './types'
// export { colors as extendColors } from './colors'
// export { default as tailwindcssPlugin } from './tailwindcss'
export function defineConfig(options?: DeepPartial<CodegenOptions>) {
  return options
}
