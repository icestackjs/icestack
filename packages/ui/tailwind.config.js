const { colors } = require('./src/colors')
// const {} = require('tailwindcss/colors')
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [],
  theme: {
    colors: {},
    extend: {
      colors: {
        ...colors,
      },
    },
  },
  plugins: [],
  presets: [],
}
