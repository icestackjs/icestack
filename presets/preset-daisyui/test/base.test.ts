import { createContext } from '@icestack/ui'

import preset from '@/index'

describe('base', () => {
  it('base css', () => {
    const ctx = createContext({
      dryRun: true,
      presets: [preset()],
    })
    const res = ctx.buildBase()
    expect(res).toMatchSnapshot()
  })
})
