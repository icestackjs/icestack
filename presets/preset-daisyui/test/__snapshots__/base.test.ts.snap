// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`base > base css 1`] = `Promise {}`;
